x/ratelimit/keeper/packet.go

```diff
@@ -102,65 +112,120 @@ func ParseDenomFromRecvPacket(packet channeltypes.Packet, packetData transfertyp
 	return denom
 }
 
-// Middleware implementation for SendPacket with rate limiting
-func (k Keeper) SendRateLimitedPacket(ctx sdk.Context, sourceChannel string, data []byte) error {
-	// The Stride channelID should always be used as the key for the RateLimit object (not the counterparty channelID)
-	// For a SEND packet, the Stride channelID is the SOURCE channel
-	// This is because the Source and Desination are defined from the perspective of a packet recipient
-	// Meaning, when this packet lands on a the host chain, the "Source" will be the Stride Channel,
-	//   and the "Destination" will be the Host Channel
-	channelId := sourceChannel
-
-	// Parse the packet data
+// Parses the sender and channelId and denom for the corresponding RateLimit object, and
+// the sender/receiver/transfer amount
+//
+// The Stride channelID should always be used as the key for the RateLimit object (not the counterparty channelID)
+// For a SEND packet, the Stride channelID is the SOURCE channel
+// For a RECEIVE packet, the Stride channelID is the DESTINATION channel
+//
+// The Source and Desination are defined from the perspective of a packet recipient
+// Meaning, when a send packet lands on a the host chain, the "Source" will be the Stride Channel,
+// and the "Destination" will be the Host Channel
+// And, when a receive packet lands on a Stride, the "Source" will be the host zone's channel,
+// and the "Destination" will be the Stride Channel
+func ParsePacketInfo(packet channeltypes.Packet, direction types.PacketDirection) (RateLimitedPacketInfo, error) {
 	var packetData transfertypes.FungibleTokenPacketData
-	if err := json.Unmarshal(data, &packetData); err != nil {
-		return err
+	if err := json.Unmarshal(packet.GetData(), &packetData); err != nil {
+		return RateLimitedPacketInfo{}, err
+	}
+
+	var channelID, denom string
+	if direction == types.PACKET_SEND {
```

Comment:
nit: would `types.PACKET_ORIGIN_STRIDE` and `types.PACKET_ORIGIN_HOST` be clearer? (don't have a strong view)

Reply:
I don't have a strong view either but I think I'd prefer to keep it as is

Reply:
sgtm!